### PR TITLE
Check for sqlite3 in automake config

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -365,7 +373,8 @@ cpp_server_ct_dns_server_LDADD = \
 	cpp/libcore.a \
 	$(evhtp_LIBS) \
 	${libevent_LIBS} \
-	-lprotobuf -lldns -lsqlite3
+	$(sqlite3_LIBS) \
```
Comment (pphaneuf) [inline Makefile.am]: This one seems spurious? Or the other below?

File: python/requirements.txt [hunk 2]
```diff
@@ -9,3 +9,4 @@ Twisted>=12.1
 bitstring
 jsonschema
 cryptography>=1.0
+pyopenssl
```
Comment (pphaneuf) [inline python/requirements.txt]: This seems unrelated?

File: configure.ac [hunk 3]
```diff
@@ -125,6 +127,14 @@ AS_IF([test -n "$missing_leveldb"],
       [AC_MSG_ERROR([could not find the leveldb/snappy libraries])])
 LIBS="$save_LIBS"
 
+save_LIBS="$LIBS"
+AS_UNSET([LIBS])
+AC_SEARCH_LIBS([sqlite3_open], [sqlite3],, [missing_sqlite3=1], [$save_LIBS])
+AC_SUBST([sqlite3_LIBS], [$LIBS])
+AS_IF([test -n "$missing_sqlite3"],
+      [AC_MSG_ERROR([could not find the sqlite3 library])])
```
Comment (pphaneuf) [inline configure.ac]: If I'm not mistaken, that just causes a message to be emitted, but doesn't stop it, right? In other words, SQLite is optional? Is that what we want? I *think* so, but just checking... :smiley:

File: python/requirements.txt (hunk 2 above)
Comment (daviddrysdale) [reply python/requirements.txt]: Yeah, I forgot to mention in the PR that there are two commits here -- this is needed to restore the Travis build to green.

File: configure.ac (hunk 3 above)
Comment (daviddrysdale) [reply configure.ac]: This will stop `./configure`, so implies SQLite is required.  Builds using gclient will always install sqlite3, so I'm hoping that's OK.

File: Makefile.am (hunk 1 above)
Comment (daviddrysdale) [reply Makefile.am]: Fixed.

File: configure.ac (hunk 3 above)
Comment (pphaneuf) [reply configure.ac]: OK, that's not what I thought, and I thought we could build without SQLite? I think I'm wrong, though, and that missing SQLite would already cause a build failure, so I think you can keep this in your PR, it might be the wrong behaviour, but it's not really different from the existing behaviour, just making it clearer, so that's still an improvement.

File: python/requirements.txt (hunk 2 above)
Comment (pphaneuf) [reply python/requirements.txt]: Might want it as a separate PR?

Comment (daviddrysdale) [reply python/requirements.txt]: Then this PR will not be mergable on its own, as Travis will be red.

(I will be doing rebase-and-merge not squash-and-merge, so it will remain a separate commit.)

File: configure.ac (hunk 3 above)
Comment (daviddrysdale) [reply configure.ac]: The existing makefiles had `-lsqlite3` hardcoded in them, so yeah, I think it was already required.

**Comment by daviddrysdale** [comment]
PTAL?

File: configure.ac (hunk 3 above)
Comment (pphaneuf) [reply configure.ac]: Yeah, I think I'm wrong on that, it was already required. Let's keep it like that for now...

File: python/requirements.txt (hunk 2 above)
Comment (pphaneuf) [reply python/requirements.txt]: I meant just this commit to `python/requirements.txt`, as its own PR, which would restore the Travis build. Then, rebase this PR on top of the now-green tree, success?

Comment (daviddrysdale) [reply python/requirements.txt]: Yeah, that would ideally make sense but it would involve 3 x Travis runs (this PR before=fail, other PR, this PR after=success) rather than one, which I wanted to avoid.
